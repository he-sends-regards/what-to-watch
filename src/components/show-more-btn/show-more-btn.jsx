import React from "react";
import PropTypes from "prop-types";
import {showMoreFilms} from "../../store/action/action";
import {connect} from "react-redux";
import {ALL_GENRE} from "../../const";

const ShowMoreBtn = ({showMoreFilmsAction, showFilmsCount, filmsLength}) => {
  return showFilmsCount < filmsLength && (
    <div className="catalog__more">
      <button className="catalog__button" type="button" onClick={() => showMoreFilmsAction()}>Show more</button>
    </div>
  );
};

ShowMoreBtn.propTypes = {
  showMoreFilmsAction: PropTypes.func.isRequired,
  filmsLength: PropTypes.number.isRequired,
  showFilmsCount: PropTypes.number.isRequired
};

const mapStateToProps = ({FILMS}) => ({
  showFilmsCount: FILMS.showFilmsCount,
  filmsLength: FILMS.films
    .filter((film) => FILMS.activeGenre === ALL_GENRE || FILMS.activeGenre === film.genre)
    .length
});

const mapDispatchToProps = (dispatch) => ({
  showMoreFilmsAction: () => {
    dispatch(showMoreFilms());
  }
});

export {ShowMoreBtn};
export default connect(mapStateToProps, mapDispatchToProps)(ShowMoreBtn);
