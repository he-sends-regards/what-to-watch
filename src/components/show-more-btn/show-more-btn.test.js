import React from "react";
import renderer from "react-test-renderer";
import {ShowMoreBtn} from "./show-more-btn";
import mockFilms from "../../mocks/films";

describe(`Should ShowMoreBtn render correctly`, () => {
  const showMoreBtnComponent = renderer.create(
      <ShowMoreBtn showMoreFilmsAction={() => {}} showFilmsCount={8} filmsLength={mockFilms.length} />
  );
  it(`Should showMoreBtnComponent render correctly`, () => {
    expect(showMoreBtnComponent.toJSON()).toMatchSnapshot();
  });
});
