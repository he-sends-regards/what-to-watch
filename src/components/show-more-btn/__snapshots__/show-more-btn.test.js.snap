// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should ShowMoreBtn render correctly Should showMoreBtnComponent render correctly 1`] = `null`;
